### PR TITLE
add restriction support like eu-access restriction

### LINKED ISSUE #232: Support Access Restrictions
**What would you like to be added**:
`Gardenctl` should add support for access restrictions. 
In case a shoot is flagged with an access restriction, `gardenctl` should notify /warn the operator before an operator accesses or starts with the troubleshooting and should inform about the access restriction. 

**How it could be configured in gardenctl**
A simplified version of the dashboard configuration for `gardenctl` could look like the following (which would need to be set for a garden configuration). This is just a proposal.
```yaml
gardenClusters:
- name: dev-virtual
  kubeConfig: path/to/kubeconfig
  accessRestrictions:
  - key: seed.gardener.cloud/eu-access
    notifyIf: true # this property is named visibleIf in the dashboard configuration
    title: EU Access # or combine title and description to `message`
    description: Do not migrate the cluster to a data center outside the EEA or Switzerland without the customer's prior written consent (email permitted). The service is offered with the regular SLAs and 24x7 support for the control plane of the cluster. 24x7 support for cluster add-ons and nodes is only available if the user complies to the conditions
    options:
    - key: support.gardener.cloud/eu-access-for-cluster-addons
      notifyIf: false
      title: EU Access Only for Cluster Addons Support # or combine title and description to `message`
      description: Do not access cluster add-ons such as DNS and certificates, Calico overlay network and network policies, kube-proxy and services, and everything else that would require direct inspection of the cluster through its API server in case you are not located in EEA. Only third-level/dev support at usual 8x5 working hours in EEA does apply in this case
    - key: support.gardener.cloud/eu-access-for-cluster-nodes
      notifyIf: false
      title: EU Access Only for Worker Node Support # or combine title and description to `message`
      description: Do not access all node-related components such as Docker and Kubelet, the operating system, and everything else that would require direct inspection of the nodes through a privileged pod or SSH in case you are not located in EEA. Only third-level/dev support at usual 8x5 working hours in EEA does apply in this case
```

**Notification to the operator**
With the above configuration, when e.g. targeting the following shoot
```yaml
apiVersion: core.gardener.cloud/v1beta1
kind: Shoot
metadata:
  annotations:
    support.gardener.cloud/eu-access-for-cluster-addons: "true"
    support.gardener.cloud/eu-access-for-cluster-nodes: "false"
  ...
spec:
  seedSelector:
    matchLabels:
      seed.gardener.cloud/eu-access: "true"
```
The follwoing notification from `gardenctl` should be shown (as defined in the `gardenctl` configuration)

---
Please Note: The following access restrictions apply for shoot _${project}/${name}_:
- **EU Access**: Do not migrate the cluster to a data center outside the EEA or Switzerland without the customer's prior written consent (email permitted). The service is offered with the regular SLAs and 24x7 support for the control plane of the cluster. 24x7 support for cluster add-ons and nodes is only available if the user complies to the conditions
  - **EU Access Only for Worker Node Support**: Do not access all node-related components such as Docker and Kubelet, the operating system, and everything else that would require direct inspection of the nodes through a privileged pod or SSH in case you are not located in EEA. Only third-level/dev support at usual 8x5 working hours in EEA does apply in this case
---

**Additional Info**:
Corresponing PR in Dashboard: https://github.com/gardener/dashboard/pull/694
Corresponing PR in Gardener: https://github.com/gardener/gardener/pull/2340

See the description of gardener/dashboard#694 for an example configuration of the dashboard.

**Why is this needed**:
An operator should be notified in case a shoot shoot is flagged with an access restriction

### PR DESCRIPTION
**Pull request by neo-liang-sap**
**What this PR does / why we need it**:
This PR will display eu-access restriction warning msg in gardenctl terminal if cluster is configed with eu-access and garden config is configed with enabling the eu-access
**Which issue(s) this PR fixes**:
Fixes https://github.com/gardener/gardenctl/issues/232

**Special notes for your reviewer**:
To use this PR:
1) shoot needs to be configed with eu-access restriction
image
2) corresponding garden config needs to be config to enable the eu-access restriction
image
3) tests has been perform for following scenario with cross permutation
shoot:
a. config eu-access
b. config node-eu-access
c. addons eu-access
garden config:
a. has/has no access restriction
b. true/false in access restriction key and option key

**Release note**:
```improvement operator
display eu-access restriction warning msg in gardenctl terminal if cluster is configed with eu-access and garden config is configed with enabling the eu-access restriction
```

### CODE REVIEW

**Comment by neo-liang-sap** [comment]
@gardener/gardenctl-maintainers any comments on this PR?

**Comment by neo-liang-sap** [comment]
per communication with @petersutter , he will review this PR after back from holiday

File: pkg/cmd/target.go [hunk 1]
```diff
@@ -996,6 +997,62 @@ func seedWrapper(targetReader TargetReader, targetWriter TargetWriter, configRea
...
+								warningMsg += option.Msg
+								warningMsg += "\n"
+							}
+						}
+
+					}
+				}
+
+			}
+		}
```
Comment (petersutter) [inline pkg/cmd/target.go:1049]: to simplify nested if statements, use early return pattern. This improves readability

**Review by petersutter — COMMENTED** [review]
overall lgtm, however to improve the readability of the code I have made some suggestions

**Comment by neo-liang-sap** [comment]
Hello @petersutter , i've fixed my PR per your comments, thanks for your review! -Neo

**Comment by petersutter** [comment]
OK I have just tested it locally. However it only worked when targeting the shoot via the dashboard url, e.g. `gardenctl target --server https://gardener.example.com --project myproject --shoot myshoot` but not via `gardenctl target shoot myshoot`

**Comment by petersutter** [comment]
I was using the command wrong for targeting a shoot by missing the `shoot` parameter. But somehow it still worked to target the shoot, except the warning was not displayed.
Why is this working anyhow? Is this intended?
```bash
bash-5.0$ ./gardenctl-darwin-amd64 target deleteme
Shoot:
KUBECONFIG=/Users/d050337/.garden/cache/dev-virtual/projects/core/deleteme/kubeconfig.yaml
                                                                                                                         
bash-5.0$ ./gardenctl-darwin-amd64 target shoot deleteme
Shoot:
KUBECONFIG=/Users/d050337/.garden/cache/dev-virtual/projects/core/deleteme/kubeconfig.yaml
Attention, EU-Access is enabled!
Do not debug cluster addons!
```

**Comment by neo-liang-sap** [comment]
Interesting. I will check the code and update here. Thanks!

**Comment by neo-liang-sap** [comment]
Hi @petersutter , thanks for your catch! yes `gardenctl target xxx` is by design and the code is [here](https://github.com/gardener/gardenctl/blob/master/pkg/cmd/target.go#L167)
i've update code to cover this scenario.
Thanks again for your nice catch!
-Neo

**Review by petersutter — APPROVED** [review]
/lgtm

**Pull request merged**